plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -135,8 +141,29 @@ public boolean advanceNextPosition()
 
         try {
             if (resultSet == null) {
-                log.debug("Executing: %s", statement.toString());
-                resultSet = statement.executeQuery();
+                try {
+                    Future<ResultSet> resultSetFuture = executor.submit(() -> {
+                        log.debug("Executing: %s", statement.toString());
+                        return statement.executeQuery();
```

Comment:
nit : What if we could move this to a wrapped implementation of `PrepatedStatement` so JDBC plugin could use them if they want. Not sure on complications bcoz of this

Reply:
You would need to have custom connection factory, custom connection, and so custom prepared statement and possibly other statements too. This invasive change that I am not able to predict the implications. 

On the other hand that approach would allow us to also cancel metadata queries when semantic analysis thread get an interruption.

CC: @findepi